openedx/core/lib/api/new_authentication.py

```diff
@@ -0,0 +1,141 @@
+""" Common Authentication Handlers used across projects. """
+
+
+import logging
+
+import django.utils.timezone
+from django.conf import settings
+from oauth2_provider import models as dot_models
+from provider.oauth2 import models as dop_models
+from rest_framework.exceptions import AuthenticationFailed
+from rest_framework.authentication import BaseAuthentication, get_authorization_header
+from edx_django_utils.monitoring import set_custom_metric
+
+
+OAUTH2_TOKEN_ERROR = 'token_error'
+OAUTH2_TOKEN_ERROR_EXPIRED = 'token_expired'
+OAUTH2_TOKEN_ERROR_MALFORMED = 'token_malformed'
+OAUTH2_TOKEN_ERROR_NONEXISTENT = 'token_nonexistent'
+OAUTH2_TOKEN_ERROR_NOT_PROVIDED = 'token_not_provided'
+
+
+logger = logging.getLogger(__name__)
+
+
+class OAuth2Authentication(BaseAuthentication):
+    """
+    This is created to be a drop in replacement for django-rest-framework-oauth oauth2Authentication class.
+    This is based on NOAuth2AuthenticationAllowINactiveUsers
+    """
+    www_authenticate_realm = 'api'
+    allow_query_params_token = settings.DEBUG
+    allow_inactive_users = False
+
+    def authenticate(self, request):
+        """
+        Returns two-tuple of (user, token) if access token authentication
+        succeeds, raises an AuthenticationFailed (HTTP 401) if authentication
+        fails or None if the user did not try to authenticate using an access
+        token.
+        """
+
+        set_custom_metric("NewOAuth2AuthenticationCalled", True)
```

Comment:
I'd rather not see `New` in this since it should be long-lived.   I'd also rather see this set up more like the other metric, with a consistent key and values for "failed", etc.